xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -4640,6 +4655,10 @@ std::string CDVDPlayer::GetPlayingTitle()
 
 bool CDVDPlayer::SwitchChannel(const CPVRChannelPtr &channel)
 {
+  const CPVRChannelPtr currentChannel(g_PVRManager.GetCurrentChannel());
+  if (channel && currentChannel && (*channel == *currentChannel))
```

Comment:
it's enough to compare the shared_ptr's because there are operators for comparing a shared_ptr with a null pointer.

```
if (channel == currentChannel)
```